addons/account/migrations/9.0.1.1/pre-migration.py

```diff
@@ -370,6 +370,38 @@ def set_date_maturity(env):
     )
 
 
+def migrate_bic_numbers(env):
+    """ OD12-343 BIC numbers not migrated in supplier bank details """
+    # if no bank exists with a matching BIC, automatically create a new bank, with the new BIC;
+    # if the bank account details do not specify the bank name, use the BIC as bank name.
+    openupgrade.logged_query(
+        env.cr, """
+        INSERT INTO res_bank(name, city, zip, country, bic, state, street, active)
+        SELECT CASE WHEN bank_name IS NOT NULL THEN bank_name ELSE bank_bic END, city, zip, country_id, bank_bic, state_id, street, active
+        FROM res_partner_bank
+        WHERE bank IS NULL AND bank_bic IS NOT NULL AND bank_bic NOT IN (
+            SELECT bic
+            FROM res_bank
+            WHERE bic IS NOT NULL
+        )
+        GROUP BY bank_name, city, zip, country_id, bank_bic, state_id, street, active
```

Comment:
We don't want multiple banks in the system with the same bic, so please group by bic alone.